internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -135,5 +145,63 @@ func (r *Receiver) createMonitor(monitorType string) (interface{}, error) {
 		output.AddExtraDimension(k, v)
 	}
 
+	collectdLock.Lock()
+	defer collectdLock.Unlock()
+	// Note, that this receiver has to configure main collectd even for collectd/custom monitor,
+	// despite the fact that, that monitor stands up its own instance of collectd to prevent this
+	// panic "Main collectd instance should not be accessed before being configured".
+	if r.config.monitorConfig.MonitorConfigCore().IsCollectdBased() && !configuredCollectd {
+		r.setUpCollectdConfig(extensions)
```

Comment:
Several monitors require specific env vars to be set by the final config to function.  Mind adding the core of https://github.com/signalfx/signalfx-agent/blob/master/pkg/core/config/config.go#L209 to this step as well?

Reply:
Updated to set `SIGNALFX_BUNDLE_DIR` and `JAVA_HOME` for now.